### PR TITLE
Implemented an ITG arrow mode to the funbox

### PR DESCRIPTION
**Pull request by SpiritAxolotl**
### Description

I have implemented ITG (in the groove) charting as a mode in "funbox"
Similar to the already-existing "arrows" mode, this one generates arrows as well, but instead of them being generated completely randomly, it has a specific generation so it's pad-compatible.

Closes #

### CODE REVIEW

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4450897102

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4450930748

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4450951583

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4451048033

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4451101965

**Comment by SpiritAxolotl** [comment]
thank you miodec

File: frontend/src/ts/test/funbox/funbox.ts [hunk 1]
```diff
@@ -169,40 +169,50 @@ FunboxList.setFunboxFunctions("arrows", {
...
+    ) {
       return true;
     }
     return false;
   },
   async preventDefaultEvent(
     event: JQuery.KeyDownEvent<Document, null, Document, Document>
   ): Promise<boolean> {
-    // TODO What's better?
-    // return /Arrow/i.test(event.key);
```
Comment (Ferotiq) [inline frontend/src/ts/test/funbox/funbox.ts:205]: I think you should keep these comments

File: frontend/src/ts/utils/misc.ts [hunk 2]
```diff
@@ -722,19 +722,68 @@ export function getASCII(): string {
...
-  for (let i = 0; i < 5; i++) {
-    let random = randomElementFromArray(arrowArray);
-    while (random === lastchar) {
-      random = randomElementFromArray(arrowArray);
+//code for "generateStep" is from Mirin's "Queue" modfile,
+// converted from lua to typescript by Spax
+// lineout: https://youtu.be/LnnArS9yrSs
+let foot = false;
+let facing = 0;
+let facingcount = 0;
```
Comment (Ferotiq) [inline frontend/src/ts/utils/misc.ts:730]: `facingcount` -> `facingCount`

File: frontend/src/ts/utils/misc.ts [hunk 3]
```diff
@@ -722,19 +722,68 @@ export function getASCII(): string {
...
+let lastl = 0,
+  lastr = 3,
+  llc = 0,
+  lrc = 0;
```
Comment (Ferotiq) [inline frontend/src/ts/utils/misc.ts:734]: Could you use more descriptive names here?

Comment (SpiritAxolotl) [reply frontend/src/ts/utils/misc.ts:734]: sure one sec

File: frontend/src/ts/utils/misc.ts [hunk 4]
```diff
@@ -722,19 +722,68 @@ export function getASCII(): string {
...
+    c = 3 - a * (facing + 1);
+  }
+
+  if (facingcount < 0 && a === 0) {
+    facing = 1 - facing;
+    facingcount = Math.floor(Math.random() * 3) + 3;
+  }
+  return c;
+}
+export function chart2Word(): string {
```
Comment (Ferotiq) [inline frontend/src/ts/utils/misc.ts:767]: newline between function declarations

File: frontend/src/ts/test/funbox/funbox.ts (hunk 1 above)
Comment (SpiritAxolotl) [reply frontend/src/ts/test/funbox/funbox.ts:205]: didn't mean to remove them whoops

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4456944341

File: frontend/src/ts/test/funbox/funbox.ts
Comment (SpiritAxolotl) [inline frontend/src/ts/test/funbox/funbox.ts:1]: thanks

File: frontend/src/ts/utils/misc.ts [hunk 5]
```diff
@@ -727,43 +727,51 @@ export function getASCII(): string {
 // lineout: https://youtu.be/LnnArS9yrSs
 let foot = false;
 let facing = 0;
-let facingcount = 0;
+let facingCount = 0;
 let lastl = 0,
   lastr = 3,
   llc = 0,
   lrc = 0;
```
Comment (Ferotiq) [inline frontend/src/ts/utils/misc.ts:734]: you forgot to delete these

Comment (SpiritAxolotl) [reply frontend/src/ts/utils/misc.ts:734]: oh whoops one sec

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4457057107

File: frontend/src/ts/utils/misc.ts [hunk 6]
```diff
@@ -725,42 +725,52 @@ export function getASCII(): string {
...
 let footTrack = false;
 let currFacing = 0;
 let facingCount = 0;
 let lastLeftStep = 0,
   lastRightStep = 3,
   leftStepCount = 0,
   rightStepCount = 0;
-function generateStep(): number {
-  let stepValue = 0;
+export function generateStep(): number {
```
Comment (Ferotiq) [inline frontend/src/ts/utils/misc.ts:736]: remove export

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/4457632349

File: frontend/src/ts/utils/misc.ts (hunk 6 above)
Comment (SpiritAxolotl) [reply frontend/src/ts/utils/misc.ts:736]: why's that?

Comment (Ferotiq) [reply frontend/src/ts/utils/misc.ts:736]: it doesn't need to be exported as you're just using it locally in `misc.ts`

Comment (SpiritAxolotl) [reply frontend/src/ts/utils/misc.ts:736]: oh good point

**Comment by SpiritAxolotl** [comment]
committed to the wrong branch

**Review by Miodec — APPROVED** [review]

**Pull request merged**